test/unit/asyncProducer.js

```diff
@@ -0,0 +1,234 @@
+/* eslint-disable import/no-extraneous-dependencies */
+const os = require('os')
+const prometheusclient = require('prom-client')
+const sap = require('supertest-as-promised')
+const sinon = require('sinon')
+const uuid = require('uuid/v4')
+const { expect } = require('chai')
+const {
+  afterEach, beforeEach, describe, it,
+} = require('mocha')
+require('co-mocha')
+
+const helpers = require('./helpers')
+const producer = require('./../../src/producer')
+const configDefault = require('./../../src/config/default.json')
+const configTest = require('./../../src/config/test.json')
+const Client = require('./../../src/client')
+
+describe('Async Producer', () => {
+  describe('Constructor', () => {
+    it('returns sync client with default config', () => {
```

Comment:
r/sync/async

Reply:
it really returns a sync client with config/default, the next test does the opposite

Reply:
If we're returning a sync client should we test it here, in the asyncProducer tests?

Reply:
we're testing that the constructor differentiates between sync/async with that particular setting (producer.async:bool), it isn't testing any particular sync behavior.

Reply:
okay, got it!